lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -337,11 +366,10 @@ sub _on_message {
     }
 
     my $missed_param;
-    if(
-        ref $message ne 'HASH' 
-        || !$message->{$missed_param = "message_id"} 
-        || !$message->{$missed_param = "response"}
-    ) {
+    if (   ref $message ne 'HASH'
+        || !$message->{$missed_param = "message_id"}
```

Comment:
Assigning to the same variable multiple times like this within the same expression is best avoided, please don't ever do this.

Instead, you can get a list of *all* the missing variables like this:

`my (@missing) = grep { !exists $message->{$_} } qw(message_id response)`

and the `ref $message ne 'HASH'` check should be split out anyway (that error makes no sense when it's not a hash).